tests/Findologic/Api/FindologicSearchServiceTest.php

```diff
@@ -170,4 +173,65 @@ public function testStagingQueryParameterWorksAsExpected(
         $isEnabled = $method->invoke($findologicSearchService, $event);
         $this->assertSame($isFindologicEnabled, $isEnabled);
     }
+
+    public function testAllowedRequestButUnknownCategoryDisablesFindologic(): void
+    {
+        $genericPageLoaderMock = $this->getMockBuilder(GenericPageLoader::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $findologicSearchService = new FindologicSearchService(
+            $this->getContainer(),
+            $this->apiClientMock,
+            $this->apiConfig,
+            $this->pluginConfigMock,
+            $genericPageLoaderMock,
+            $this->getContainer()->get(SortingService::class),
+            $this->getContainer()->get(PaginationService::class)
+        );
+
+        $genericPageLoaderMock->expects($this->once())->method('load')->willreturn(new Page());
+
+        $this->pluginConfigMock->expects($this->any())->method('isInitialized')->willReturn(true);
+
+        $findologicService = new FindologicService();
+        $findologicService->enable();
+        $findologicService->enableSmartSuggest();
+
+        $contextMock = $this->getMockBuilder(Context::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $contextMock->expects($this->any())->method('getExtension')->willReturnCallback(
+            function (string $name) use ($findologicService) {
+                if ($name === 'findologicService') {
+                    return $findologicService;
+                }
+
+                return null;
+            }
+        );
+
+        $request = new Request(['navigationId' => $this->getRootCategory()->getId()]);
```

Comment:
So root category is the unknown one, how does the FindologicService identify categories?

Reply:
The class `\FINDOLOGIC\FinSearch\Findologic\Request\Parser\NavigationCategoryParser` is responsible for finding out which category the current category is. The root category is considered unknown, as this is a category, which is not directly indexed by Findologic.

Reply:
Please add this as a comment, as it's not obvious.